puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/tarballsRelease.groovy

```diff
@@ -0,0 +1,94 @@
+pipeline {
+    agent { label 'admin && sshkey' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        disableConcurrentBuilds()
+        ansiColor('xterm')
+    }
+
+    def version = env.getProperty('version')
+    def major_version = env.getProperty('major_version')
+    def ruby_ver = '2.4.0'
+
+    stages {
+        stage('Parallel') {
+
+            parallel {
+                stage('Foreman') {
+                    steps {
+                        verify_tag('foreman', version)
+                        build_tarball('foreman', version, ruby_ver)
+                    }
+                }
+                stage('Smart Proxy') {
+                    steps {
+                        verify_tag('smart-proxy', version)
+                        build_tarball('smart-proxy', version, ruby_ver)
+                    }
+                }
+                stage('Foreman SELinux') {
+                    steps {
+                        verify_tag('foreman-selinux', version)
+                        build_tarball('foreman-selinux', version, ruby_ver)
+                    }
+                }
+                stage('Foreman Installer') {
+                    steps {
+                        verify_tag('foreman-installer', version)
+                        build_tarball('foreman-installer', version, ruby_ver)
+                    }
+                }
+            }
+        }
+    }
+
+    post {
+        success {
+            build job: 'release_packages', parameters: [
+                [$class: 'StringParameterValue', name: 'version', value: version],
+                [$class: 'StringParameterValue', name: 'major_version', value: major_version]
+            ]
+        }
+    }
+}
+
+void verify_tag(project, version) {
+    dir(project) {
+        git url: "https://github.com/theforeman/${project}", branch: 'develop'
+        sh "test \"$(git tag -l ${version} | wc -l)\" -eq 1"
+    }
+}
+
+void build_tarball(project, version, ruby_ver) {
+    def base_dir = "/var/www/vhosts/downloads/htdocs/foreman"
+    def rake = "rake"
+
+    dir(project) {
+        checkout scm: [$class: 'GitSCM',
+            userRemoteConfigs: [[url: "git@github.com:theforeman/${project}.git"]],
```

Comment:
Would it be better to use ```https://```? Not sure if we should rely on SSH keys if there's no need for it.

Reply:
updated